### ISSUE TITLE
Update pummelzacken BIOS & IPMI

### ISSUE THREAD

**Issue by zerebubuth** [openstreetmap/operations#32]
Pummelzacken's [BIOS upgrade](http://www.supermicro.nl/products/motherboard/Xeon/C600/X9DRW-iF.cfm) is worth installing. @Firefishy also suggests installing the IPMI update at the same time. (link?) By far the easiest is to make a USB boot USB using something like: http://rufus.akeo.ie/ and visiting to install.

**Comment by tomhughes** [comment]
Also the RAID controller firmware: http://www.adaptec.com/en-us/downloads/bios_fw/bios_fw_ver/productid=asr-81605zq&dn=adaptec+raid+81605zq.php

**Comment by tomhughes** [comment]
https://gist.github.com/tomhughes/b30029b16ccc8198e3c2 is the RAID controller lockup we're hoping to fix,

**Comment by zerebubuth** [comment]
http://www.bhargavs.com/index.php/2014/06/13/taking-steps-backwards-update-bios-on-supermicro-servers-with-dos-bootable/ is a guide to updating the BIOS. @Firefishy recommends using the web interface to update IPMI.

**Comment by zerebubuth** [comment]
BIOS & RAID controller firmwares updated. IPMI's loader (`RKCSFLSH.EXE`) appears to do nothing more than immediately reboot the machine), so we'll try a remote load of that over the web interface.

**Comment by Firefishy** [comment]
I have updated the IPMI firmware from within linux and confirmed working.

**Comment by lonvia** [comment]
The performance of pummelzacken has significantly decreased since the update to the point of making the machine unusable. I've disabled updates and put heavy banning restrictions in place but it barely makes a difference. A rogue client might be involved as well but altogether it looks like a problem of IO throughput.

**Comment by tomhughes** [comment]
I'm not sure it correlates to the update does it? I noticed a big change last night, but that was a few hours after it was updated?

**Comment by tomhughes** [comment]
Specifically it seemed like performance fell off a cliff around 8pm UTC some three and a half hours after the machine was bought back up.

I actually thought I had caused it to start with, as it seemed to correlate with my trying to fix the streaming replication server, although I couldn't see how and I eventually ruled that out.

**Comment by lonvia** [comment]
I switched updates back on at this point, so postgres writes start at this point. Switching off updates helped later for a bit (the dip around 22:30) but not for long. Currently postgres is vacuuming, so there is still write IO ongoing.

**Comment by lonvia** [comment]
The streaming-replication change looks unrelated but I wonder if the reboot has sneaked in other configuration changes for postgres or similar.

**Comment by lonvia** [comment]
FYI, the issue reappeared yesterday after pummelzacken had spontaneously rebooted. The software raid for the root partition had to be rebuilt for some reason and half-way through write speed slowed down by a factor of 30 and did not recover after the rebuild was finished. I didn't further investigate if the hardware or the driver is to blame or if the reboot was related.

I've moved the postgres data directory to the /data partition which hopefully avoids further interference with operations of Nominatim.

**Comment by Firefishy** [comment]
@zerebubuth has been capturing a kernel oop log for the last few months: ridley:/home/matt/screenlog.0
All the kernel panics have aac_intr_normal in the top of the stack trace.

I have now updated the aacraid module to the latest vendor driver for adaptec via dkms. I have also installed linux-generic-lts-wily (4.2.0.x) and specifically excluded it from aacraid dkms.

System requires a reboot to activate.
